Add printer profile for Modix3D printers

ComedianTF2 opened the pull request:
Has printer profiles for the Modix V3 line of printers: the Big40, 60, 120X, 120Z, 180X and Meter

nallath commented on resources/definitions/modix_base.def.json:
```diff
@@ -0,0 +1,198 @@
+        "preferred_variant_name": "0.4 mm Nozzle",
+		"preferred_material": "generic_pla",
+		"exclude_materials": [
+            "ultimaker_abs_black",
```
No need to exclude these, they are 2.85 mm materials.

nallath commented on resources/definitions/modix_base.def.json:
```diff
@@ -0,0 +1,198 @@
+        "filter_out_tiny_gaps": { "value": false },
+
+        "retraction_speed": {
+            "maximum_value_warning": "machine_max_feedrate_e if retraction_enable else float('inf')",
```
```suggestion
            "maximum_value_warning": "machine_max_feedrate_e if retraction_enable else float('inf')",
```
No need to set this; The default is already the same (DRY: Don't Repeat Yourself)

nallath replied:
There are other cases where this is also the case. I've not marked all of them.

ComedianTF2 commented:
Please hold on, I made some mistakes in that commit and need to fix it still

ComedianTF2 commented:
Updated the modix_base.def.json to delete any duplicates from the fdmprinter.def.json

nallath commented:
The version number of the profiles isn't correct. THat is why the tests are failing. It should be 19

ComedianTF2 commented:
Fixed the version number from Version 2 to Version 19

nallath commented:
Nope, sorry. Not the definition one. That should be 2. The version number for the profiles (setting_version = 17) should be 19.

ComedianTF2 commented:
> Nope, sorry. Not the definition one. That should be 2. The version number for the profiles (setting_version = 17) should be 19.

Fixed the profiles version from 17 to 19, fixed the base definition from 19 to 2

nallath commented:
The CI is still failing on some things.

 [MainThread] UM.Settings.DefinitionContainer._preprocessParsedJson [299]: Unable to override setting travel_compensate_overlapping_walls_0_enabled
[MainThread] UM.Settings.DefinitionContainer._preprocessParsedJson [299]: Unable to override setting fill_perimeter_gaps

Unnecessary default_values for ['retraction_prime_speed', 'retraction_retract_speed'] in /__w/Cura/Cura/tests/Settings/../../resources/definitions/modix_base.def.json

ComedianTF2 commented:
@nallath I'm a little confused about the retraction_prime_speed and retraction_retract_speed, as the default values in fdmprinter.def.json are set to 25, rather than the preferred 40.

nallath commented:
> @nallath I'm a little confused about the retraction_prime_speed and retraction_retract_speed, as the default values in fdmprinter.def.json are set to 25, rather than the preferred 40.

Correct. But the _value_ property of these is also set. So if you only change the default_value it won't actually change anything. If the value is set, that is what is used (and not the default_value).

ComedianTF2 commented:
@nallath Do you know what's causing the issue with the failing check?

ComedianTF2 commented:
@Ghostkeeper Thanks for the info. The 5.0 launch looks very exciting and I understand the crunch taking priority.I'll be re-submitting the profile again tomorrow, hopefully it will work properly.
